pkg/controllers/dynakube/activegate/capability/capability.go

```diff
@@ -206,3 +207,34 @@ func BuildProxySecretName(dynakubeName string) string {
 func BuildServiceName(dynakubeName string, module string) string {
 	return dynakubeName + "-" + module
 }
+
+// BuildServiceHostName converts the name returned by BuildServiceName
+// into the variable name which Kubernetes uses to reference the associated service.
+// For more information see: https://kubernetes.io/docs/concepts/services-networking/service/
+func BuildServiceHostName(dynakubeName string, module string) string {
+	serviceName := BuildServiceNameUnderscore(dynakubeName, module)
+	return fmt.Sprintf("$(%s_SERVICE_HOST):$(%s_SERVICE_PORT)", serviceName, serviceName)
+}
+
+// BuildServiceDomainName builds service domain name
+func BuildServiceDomainName(dynakubeName string, namespaceName string, module string) string {
+	return fmt.Sprintf("%s.%s:$(%s_SERVICE_PORT)", BuildServiceName(dynakubeName, module), namespaceName, BuildServiceNameUnderscore(dynakubeName, module))
+}
+
+// BuildServiceNameUnderscore converts result of BuildServiceName by replacing dashes with underscores
+// to make it env variable compatible because it's only special symbol it supports
+func BuildServiceNameUnderscore(dynakubeName string, module string) string {
+	return strings.ReplaceAll(
+		strings.ToUpper(
+			BuildServiceName(dynakubeName, module)),
+		"-", "_")
+}
+
+// BuildDNSEntryPoint for give capability
+func BuildDNSEntryPoint(dynakubeName, dynakubeNamespace string, capability Capability) string {
+	if capability.ShortName() == consts.MultiActiveGateName && strings.Contains(capability.ArgName(), dynatracev1beta1.RoutingCapability.ArgumentName) ||
+		capability.ShortName() == dynatracev1beta1.RoutingCapability.ShortName {
+		return fmt.Sprintf("https://%s/communication,https://%s/communication", BuildServiceHostName(dynakubeName, capability.ShortName()), BuildServiceDomainName(dynakubeName, dynakubeNamespace, capability.ShortName()))
+	}
+	return fmt.Sprintf("https://%s/communication", BuildServiceHostName(dynakubeName, capability.ShortName()))
+}
```

Comment:
Using this logic for the codemodules is a major flaw, and it will not work

So the PR works as far as it updates the ruxitagentproc conf and all:
```
bash-5.1# cat data/zib50933/config/agent/conf/ruxitagentproc.conf  | grep noProxy
noProxy https://$(ZIB50933_ACTIVEGATE_SERVICE_HOST):$(ZIB50933_ACTIVEGATE_SERVICE_PORT)/communication,https://zib50933-activegate.dynatrace:$(ZIB50933_ACTIVEGATE_SERVICE_PORT)/communication
```

however these environment variables:
- ZIB50933_ACTIVEGATE_SERVICE_HOST 
- ZIB50933_ACTIVEGATE_SERVICE_PORT
- ...
will NOT be present in the user's namespace, where the codemodules will run as the service is not in there namespace so kubernetes will not add these envvars automatically.
What is even worse, is that we couldn't even just evaluate these envvars in the CSI-driver/Operator as the Service is created dynamically, and these are envvars are only added to the container if the Service already existed when the pod was scheduled.
[official docs for service envvars](https://kubernetes.io/docs/concepts/services-networking/service/#environment-variables)